### PR TITLE
Add OpenStack Observability DT

### PR DESCRIPTION
**Pull request by myadla**
Created DT for telemetry-operator in cloudops

### CODE REVIEW

File: dt/cloudops/openstack-telemetry_DT.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+| Controller nodes                              | vm    | 2      |
+
+
+## Services, enabled features and configurations
+| Service                                     | configuration                     | Lock-in coverage?  |
+| ------------------------------------------- | ----------------------------------| ------------------ |
+| Nova                                        | default                           | Standard           |
+| Neutron                                     | default                           | Standard           |
+| Cinder                                      | default                           | Standard           |
+| Glance                                      | NetApp, Fibre channel             | Standard           |
```
Comment (mrunge) [inline dt/cloudops/openstack-telemetry_DT.md:24]: config can be default, we have no specific requirement

File: dt/cloudops/openstack-telemetry_DT.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+## General information
+
+| Revision | Change                | Date             |
+|--------: | :-------------------- | :--------------: |
+| v0.1     | Initial publication   | 14.12.23      |
+
+## Node topology
+| Node role                                     | bm/vm | amount |
+| --------------------------------------------- | ----- | ------ |
+| Compute nodes                                 | vm    | 3      |
```
Comment (mrunge) [inline dt/cloudops/openstack-telemetry_DT.md:14]: 2 compute nodes

File: dt/cloudops/openstack-telemetry_DT.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+| Controller nodes                              | vm    | 2      |
```
Comment (mrunge) [inline dt/cloudops/openstack-telemetry_DT.md:15]: 3 controller nodes

File: dt/cloudops/openstack-telemetry_DT.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# RHOSO Deployed Topology DFG-cloudops-metmon-openstack-telemetry-18.0-3cont_2comp_final-ipv4-rhosfips
```
Comment (mrunge) [inline dt/cloudops/openstack-telemetry_DT.md:1]: the naming suggests 3 controllers, 2 compute nodes and enabling fips (which *should* be fine).

**Review by mrunge — CHANGES_REQUESTED** [review]
Hi Muneesha, thank you for working on this!
I have a few nits

File: dt/cloudops/openstack-telemetry_DT.md (hunk 4 above)
Comment (tosky) [reply dt/cloudops/openstack-telemetry_DT.md:1]: about the naming, please check the changes to the default template.  I would also use a less, uhm, detailed, name (please check the other proposals).

Comment (mrunge) [reply dt/cloudops/openstack-telemetry_DT.md:1]: let's use something like observbility for the name and not use a name that would imply a certain configuration (like number of controllers, or fips enabled/disabled).

File: dt/cloudops/openstack-telemetry_DT.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+## Considerations/Constraints
+1. Telemetry should be enabled that will enable autoscaling, telemetry, prometheus deployment, grafana deployment, etc.
+2. No topology constraints
+
+
+## Testing tree
+
+| Test framework       | Stage to run  | Special configuration | Test case to report |
+| -------------------- | ------------  | --------------------- | :-----------------: |
+| Tempest              | stage 10      | Use rhel image        |             |
```
Comment (mrunge) [inline dt/cloudops/openstack-telemetry_DT.md:37]:
```suggestion
| Tempest              | stage 10      | Use rhel image        |             |tempest-telemetry
```

File: dt/cloudops/openstack-telemetry_DT.md (hunk 4 above)
Comment (mrunge) [inline dt/cloudops/openstack-telemetry_DT.md:1]:
```suggestion
# RHOSO Deployed Topology OpenStack Observability
```

File: dt/cloudops/openstack-telemetry_DT.md (hunk 1 above)
Comment (myadla) [reply dt/cloudops/openstack-telemetry_DT.md:24]: Done

File: dt/cloudops/openstack-telemetry_DT.md (hunk 2 above)
Comment (myadla) [reply dt/cloudops/openstack-telemetry_DT.md:14]: Done

File: dt/cloudops/openstack-telemetry_DT.md (hunk 3 above)
Comment (myadla) [reply dt/cloudops/openstack-telemetry_DT.md:15]: Done

File: dt/cloudops/openstack-telemetry_DT.md (hunk 4 above)
Comment (myadla) [reply dt/cloudops/openstack-telemetry_DT.md:1]: Done

Comment (myadla) [reply dt/cloudops/openstack-telemetry_DT.md:1]: Done

File: dt/cloudops/openstack-telemetry_DT.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (myadla) [reply dt/cloudops/openstack-telemetry_DT.md:37]: Done

File: dt/cloudops/openstack-telemetry_DT.md (hunk 4 above)
Comment (myadla) [reply dt/cloudops/openstack-telemetry_DT.md]: @mrunge @tosky I have committed the changes

**Review by mrunge — APPROVED** [review]
thank you, looks good to  me

**Comment by tosky** [comment]
Please merge the comments and change the commit message to really explain what this is about, without references to an internal issue.

File: dt/cloudops/openstack-telemetry_DT.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+| ------------------------------------------- | ----------------------------------| ------------------ |
+| Nova                                        | default                           | Standard           |
+| Neutron                                     | default                           | Standard           |
+| Cinder                                      | default                           | Standard           |
+| Glance                                      | default                           | Standard           |
+| Keystone                                    | default                           | Standard           |
+| telemetry                                   | aka ceilometer, aodh & prometheus | Must have          |
+
+## Considerations/Constraints
+1. Telemetry should be enabled that will enable autoscaling, telemetry, prometheus deployment, grafana deployment, etc.
```
[the remark this answers is not in the document]
Comment (jlarriba) [reply dt/cloudops/openstack-telemetry_DT.md:29]: That still needs to be discussed, we might be including grafana deployment on the operator

File: dt/cloudops/openstack-telemetry_DT.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+## Services, enabled features and configurations
+| Service                                     | configuration                     | Lock-in coverage?  |
+| ------------------------------------------- | ----------------------------------| ------------------ |
+| Nova                                        | default                           | Standard           |
+| Neutron                                     | default                           | Standard           |
+| Cinder                                      | default                           | Standard           |
+| Glance                                      | default                           | Standard           |
+| Keystone                                    | default                           | Standard           |
+| telemetry                                   | aka ceilometer, aodh & prometheus | Must have          |
```
Comment (jlarriba) [inline dt/cloudops/openstack-telemetry_DT.md:26]: This should be with a capital 'T': Telemetry

File: dt/cloudops/openstack-telemetry_DT.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+
+## Services, enabled features and configurations
+| Service                                     | configuration                     | Lock-in coverage?  |
+| ------------------------------------------- | ----------------------------------| ------------------ |
+| Nova                                        | default                           | Standard           |
+| Neutron                                     | default                           | Standard           |
+| Cinder                                      | default                           | Standard           |
+| Glance                                      | default                           | Standard           |
+| Keystone                                    | default                           | Standard           |
```
Comment (jlarriba) [inline dt/cloudops/openstack-telemetry_DT.md:25]: I dont know if Keystone already deploys them, but Telemetry depends on Memcached and RabbitMQ.

We should define Memcached, RabbitMQ and Keystone as Must have due to prerequisites.

**Comment by myadla** [comment]
Updated the PR

**Review by jlarriba — APPROVED** [review]

File: examples/dt/cloudops/openstackcontrolplane.yaml [hunk 9]
```diff
@@ -0,0 +1,391 @@
...
+  telemetry:
+    enabled: true
+    template:
+      metricStorage:
+        enabled: true
+        monitoringStack:
+          type: prometheus
+      autoscaling:
+        enabled: true
+        prometheus:
```
[the remark this answers is not in the document]
Comment (jlarriba) [reply examples/dt/cloudops/openstackcontrolplane.yaml:324]: Yes, you are right. Fixed.

File: examples/dt/cloudops/openstack-telemetry_DT.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+1. Telemetry should be enabled that will enable autoscaling, telemetry, prometheus deployment, grafana deployment, etc.
+2. No topology constraints
+
+
+## Testing tree
+
+| Test framework       | Stage to run  | Special configuration | Test case to report |
+| -------------------- | ------------  | --------------------- | :-----------------: |
+| Tempest-telemetry    | stage 10      | Use rhel image        |    TBD         |
+| Autscaling           | stage 11      | Use cirros image      |    TBD         |
```
Comment (leifmadsen) [inline examples/dt/cloudops/openstack-telemetry_DT.md:40]:
```suggestion
| Auto scaling         | stage 11      | Use cirros image      |    TBD         |
```

File: examples/dt/cloudops/openstack-telemetry_DT.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+| STF Functional tests | stage 12      | Use cirros image      |    TBD         |
```
Comment (leifmadsen) [inline examples/dt/cloudops/openstack-telemetry_DT.md:41]: Is this correct? Is the target version here OSP 18? If so, I think STF functional testing is out of scope? Perhaps this is planned to be backported or something?

Comment (jlarriba) [reply examples/dt/cloudops/openstack-telemetry_DT.md]: No it is not. We are not running STF functional tests for RHOSO, so I have removed it from the document.

File: examples/dt/cloudops/openstack-telemetry_DT.md (hunk 10 above)
Comment (jlarriba) [reply examples/dt/cloudops/openstack-telemetry_DT.md]: It should be Autoscaling, without space. I have fixed it.

**Comment by cmsj** [comment]
Does this make sense as its own DT? I thought the intention was to try and collect as much stuff as possible into existing DTs so they can test lots of stuff all at once.

**Comment by leifmadsen** [comment]
> > Does this make sense as its own DT? I thought the intention was to try and collect as much stuff as possible into existing DTs so they can test lots of stuff all at once.
> 
> As far as we discussed, it does not make a lot of sense for Telemetry to have its own DT. It would add much more value to test Telemetry as part of a different DT that also runs workloads. Moreover, Telemetry will be enabled on most VAs, and that also will get it tested.
> 
> Any opinions against closing this PR?

I'd be ok with closing this standalone DT for telemetry, as long as there are some PR(s) that might show up to extend existing DTs so we don't forget to enable telemetry at the base.

**Pull request closed without merging**